core/domain/exp_domain.py

```diff
@@ -738,15 +738,15 @@ def validate(self, strict=False):
                 % self.param_changes)
         for param_change in self.param_changes:
             param_change.validate()
-            if param_change.name not in self.param_specs:
```

Comment:
@lilithxxx -- here and elsewhere, why are we deleting code in the non-test files? I think we already discussed about not doing this unless there's a watertight reason for doing so, in which case it should be called out explicitly in a comment.

/cc @brianrodri

Reply:
Sorry but its similar reason as here: https://github.com/oppia/oppia/pull/6844/files/20c26c4c571e4c60d6dc284df2e6c2e4d8a32565#r300104037

I did not delete this code, just shifted it after the `if param_change.name in feconf.INVALID_PARAMETER_NAMES` check(line 741). This is because we cannot test for INVALID_PARAMETER_NAMES as it cannot be in self.param_changes.